compiler/circle-eval-diff/src/InputDataLoader.h

```diff
@@ -73,6 +74,21 @@ class HDF5Loader final : public InputDataLoader
   std::unique_ptr<dio::hdf5::HDF5Importer> _hdf5;
 };
 
+// This class loads the directory that has raw data binary files.
+class DirectoryLoader final : public InputDataLoader
+{
+public:
+  DirectoryLoader(const std::string &dir_path, const std::vector<loco::Node *> &input_nodes);
+
+public:
+  uint32_t size(void) const final;
+  Data get(uint32_t data_idx) const final;
+
+private:
+  const std::vector<loco::Node *> _input_nodes;
+  std::vector<Data> _data_set;
```

Comment:
It seems that all data are loaded at once. Could you change this to load data one by one whenever each is used? AFAIK, the current interface (`size`, `get`) was introduced for that purpose. One possible implementation would be as follows.

1. Save file names of the data in constructor. (for example, in `std::vector<std::string> _data_files`)
2. When `get` is called, open the file for the given index and read data.